src/gprs.c

```diff
@@ -1984,9 +1984,10 @@ static DBusMessage *gprs_set_property(DBusConnection *conn,
 					gprs->settings);
 		}
 
+		notify_powered_change(gprs);
+
 		gprs_netreg_update(gprs);
 
-		notify_powered_change(gprs);
```

Comment:
notify_powered_change() does things only when we power on data for a modem. Changing the order here makes sure that we first detach the other modems before trying to attach for the modem, which makes more sense than the other way around (we are "freeing" the radio resources before trying to attach from a different slot).